compiler/rustc_trait_selection/src/solve/trait_goals.rs

```diff
@@ -188,6 +188,16 @@ impl<'tcx> assembly::GoalKind<'tcx> for TraitPredicate<'tcx> {
         })
     }
 
+    /// ```rust,ignore (not valid rust syntax)
+    /// impl Sized for u*, i*, bool, f*, FnPtr, FnDef, *(const/mut) T, char, &mut? T, [T; N], dyn* Trait, !
+    ///
+    /// impl Sized for (T1, T2, .., Tn) where T1: Sized, T2: Sized, .. Tn: Sized
+    ///
+    /// impl Sized for Adt where T: Sized forall T in field types
+    /// ```
+    ///
+    /// note that `[T; N]` is unconditionally sized since `T: Sized` is required for the array type to be
+    /// well-formed.
```

Comment:
My initial reaction was that I would prefer moving the comment for each builtin impl to the most specific place possible, having only a single impl per comment. This would mostly correspond to commenting the match arms on the `self_ty` matches in most cases.

I worry that these comments here can easily get out of date. E.g. When changing `instantiate_constituent_tys_for_sized_trait` we likely won't remember updating this function :thinking: E.g. this function currently diverges from the old solver by checking that all tuple fields are sized, not only the last one

Reply:
Sure, that would work. I will move these.